Framework/DataObjects/src/Histogram1D.cpp

```diff
@@ -65,5 +65,18 @@ const MantidVec &Histogram1D::dataDx() const { return m_histogram.dataDx(); }
 /// Deprecated, use dx() instead.
 const MantidVec &Histogram1D::readDx() const { return m_histogram.readDx(); }
 
+/**
+ * Makes sure the histogram has valid Y data.
+ * @param histogram A histogram to process.
+ * @throw std::invalid_argument if Y data is NULL.
+ */
+void Histogram1D::checkAndSanitizeHistogram(
+    HistogramData::Histogram &histogram) {
+  if (!histogram.sharedY()) {
```

Comment:
Would this test be better in `ISpectrum` so all histograms get it by default (unless they override)?
Currently we do no checking here:
https://github.com/mantidproject/mantid/blob/master/Framework/API/inc/MantidAPI/ISpectrum.h#L246

Additionally what about E data being null?